### ISSUE TITLE
SignInWithIdToken not returning updated session

### ISSUE THREAD

**Issue by kaushalkumar86** [supabase-community/gotrue-csharp#89]
when called

Session session = await supabase.Auth.SignInWithIdToken(Constants.Provider.Google, task.Result.IdToken);

the resultant session does not contain vales for Google provider, that is, the session value itself is not null but session.User.Identities does not contain anything, its count values is 0.

Have to use 
supabase.Auth.LoadSession();
Session session = await supabase.Auth.RetrieveSessionAsync();

after signin to get proper updated session details with User.Identities to check for provider and its data.

**Comment by acupofjose** [comment]
Can you check with v4.3.0 as to whether this issue has been resolved?

**Comment by kaushalkumar86** [comment]
Will do and let you know today

**Comment by kaushalkumar86** [comment]
Yes its working now and I am getting the updated session, thanks a lot.
But, I can see that you also added native login support for Facebook, that is, in SignInWithIdToken you have added facebook support. Earlier on firebase, I used Facebook SDK to login and get access_token and exchange that token to sign in to firebase.

In supabase, it was not supported for facebook, but for google SignInWithIdToken it is working. 

Now since you have also added facebook in SignInWithIdToken, and I tried, I am getting error. Facebook SDK does not provide any id_token, it only provided access_token. 
When I tried sending the value of access_token in both id_token and access_token, I got the error as bad id_token, that is SignInWithIdToken(facebook, accesToken, accessToken).
When tried to send null as id_token, the error was id_token is required, that is SignInWithIdToken(facebook, null, acessToken).
Please tell me, what am I missing something?

Right now, I have a java(aar) plugin for chrome tab (for android) to use it with Unity in order to signin to facebook via PKCE, its working. Then i had to call fb graph to get my access_token to that I can get user extra data like friends and all, it also works.
But the issue with this method is, I am limited to platform and dependent on chrome to be installed on the device.

**Comment by kaushalkumar86** [comment]
Thanks for the update,

I have contacted @wiverson last year, suggested him proguard rules and
continous  networkstatus check for games which need continuousconnection to
internet.
With the help of his video, I was able to successfully switch to supabase
and implement native Google signin with sdk.
Already using PKCE flow with deeplinking to signin to supabase.
Your docs are self explanatory and very helpful.
